Mathlib/Analysis/SpecialFunctions/Log/PosLog.lean

```diff
@@ -0,0 +1,160 @@
+scoped notation "log⁺" => posLog
+
+/-- Definition of the positive part of the logarithm, formulated as a theorem. -/
+theorem posLog_def {r : ℝ} : log⁺ r = max 0 (log r) := rfl
+
+/-!
+## Elementary Properties
+-/
+/-- Presentation of `log` in terms of its positive part. -/
+theorem posLog_sub_posLog_inv {r : ℝ} : log⁺ r - log⁺ r⁻¹ = log r := by
+  rw [posLog_def, posLog_def, log_inv]
+  by_cases h : 0 ≤ log r
+  · simp [h]
+  · rw [not_le] at h
+    simp [neg_nonneg.1 (Left.nonneg_neg_iff.2 h.le)]
+
+/-- Presentation of `log⁺` in terms of `log`. -/
+theorem half_mul_log_add_log_abs {r : ℝ} : 2⁻¹ * (log r + |log r|) = log⁺ r := by
+  by_cases hr : 0 ≤ log r
+  · simp [posLog, hr, abs_of_nonneg]
+    ring
+  · simp [posLog, le_of_not_ge hr, abs_of_nonpos]
+
+/-- The positive part of `log` is never negative. -/
+theorem posLog_nonneg {x : ℝ} : 0 ≤ log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is even. -/
+@[simp]
+theorem posLog_neg (x : ℝ) : log⁺ (-x) = log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is even. -/
+theorem posLog_abs (x : ℝ) : log⁺ |x| = log⁺ x := by simp [posLog]
+
+/-- The function `log⁺` is zero in the interval [-1,1]. -/
+theorem posLog_eq_zero_iff (x : ℝ) : log⁺ x = 0 ↔ |x| ≤ 1 := by
+  rw [← posLog_abs, ← log_nonpos_iff (abs_nonneg x)]
+  simp [posLog]
+
+/-- The function `log⁺` equals `log` outside of in the interval (-1,1). -/
+theorem posLog_eq_log {x : ℝ} (hx : 1 ≤ |x|) : log⁺ x = log x := by
+  simp [posLog]
+  rw [← log_abs]
+  apply log_nonneg hx
+
+/-- The function `log⁺` equals `log` for all natural numbers. -/
+theorem log_of_nat_eq_posLog {n : ℕ} : log⁺ n = log n := by
+  by_cases hn : n = 0
+  · simp [hn, posLog]
+  · simp [posLog_eq_log, Nat.one_le_iff_ne_zero.2 hn]
+
+/-- The function `log⁺` is monotone in the positive axis. -/
+theorem monotoneOn_posLog : MonotoneOn log⁺ (Set.Ici 0) := by
+  intro x hx y hy hxy
+  simp [posLog]
+  by_cases h : log x ≤ 0
+  · tauto
+  · right
+    have := log_le_log (lt_trans Real.zero_lt_one ((log_pos_iff hx).1 (not_le.1 h))) hxy
+    simp [this]
+    linarith
+
+/-!
+## Estimates for Products
+-/
+/-- Estimate for `log⁺` of a product. See `Real.posLog_prod` for a variant involving
+multiple factors. -/
+theorem posLog_mul {a b : ℝ} :
+    log⁺ (a * b) ≤ log⁺ a + log⁺ b := by
+  by_cases ha : a = 0
+  · simp [ha, posLog]
+  by_cases hb : b = 0
+  · simp [hb, posLog]
+  unfold posLog
+  nth_rw 1 [← add_zero 0, log_mul ha hb]
+  exact max_add_add_le_max_add_max
+
+/-- Estimate for `log⁺` of a product. Special case of `Real.posLog_mul` where one of
+the factors is a natural number. -/
+theorem posLog_nat_mul {n : ℕ} {a : ℝ} :
+    log⁺ (n * a) ≤ log n + log⁺ a := by
+  rw [← log_of_nat_eq_posLog]
+  exact posLog_mul
+
+/-- Estimate for `log⁺` of a product. See `Real.posLog_mul` for a variant with
+only two factors. -/
+theorem posLog_prod {α : Type} [DecidableEq α] (s : Finset α) (f : α → ℝ) :
+    log⁺ (∏ t ∈ s, f t) ≤ ∑ t ∈ s, log⁺ (f t) := by
```

Comment:
```suggestion
theorem posLog_prod {α : Type*} (s : Finset α) (f : α → ℝ) :
    log⁺ (∏ t ∈ s, f t) ≤ ∑ t ∈ s, log⁺ (f t) := by
  classical
```
No need for the decidability assumption in the statement: you just need it for the proof, so use can use `classical`, which tells to Lean: don't bother me with these silly things, I am a mathematician.